src/SignalR/server/SignalR/test/Internal/DefaultHubDispatcherTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests.Internal
+{
+    public class DefaultHubDispatcherTests
+    {
+        private class MockHubConnectionContext<TValue> : HubConnectionContext
```

Comment:
The simplest way I could think to validate the values that get dispatched to the hub was via the HubConnectionContext, capturing the responses it receives. (The behaviour is buried a little deep inside the dispatcher.)